src/registrar/models/domain_request.py

```diff
@@ -525,13 +539,40 @@ def sync_organization_type(self):
         # Actually updates the organization_type field
         org_type_helper.create_or_update_organization_type()
 
+    def _cache_status_and_action_needed_reason(self):
```

Comment:
Blocking: This would fail if you load a fresh session, no? We should check logs instead of relying on cache.

Reply:
Nvm, I see this fires on init.

Do we need to do the same for rejection reason? (out of scope)